diff --git a/src/components/Chat/ChatInput.vue b/src/components/Chat/ChatInput.vue
index d6ee132..ae4d141 100644
--- a/src/components/Chat/ChatInput.vue
+++ b/src/components/Chat/ChatInput.vue
@@ -8,6 +8,7 @@ export interface ChatInputProps {
 
 export type ChatInputEmits = {
   send: [content: string]
+  abort: []
 }
 
 const props = defineProps<ChatInputProps>()
@@ -15,14 +16,11 @@ const emit = defineEmits<ChatInputEmits>()
 
 const text = ref('')
 
-const canSend = computed(() => {
-  const isNonEmpty = text.value.trim().length > 0
-
-  return isNonEmpty && !props.isProcessing
-})
+const canSend = computed(() => text.value.trim().length > 0)
 
 function send() {
   if (!canSend.value) return
+  if (props.isProcessing) return
 
   emit('send', text.value.trim())
 
@@ -35,6 +33,14 @@ function onCtrlEnter(evt: KeyboardEvent) {
 
   send()
 }
+
+function handleAbort() {
+  if (!props.isProcessing) {
+    return
+  }
+
+  emit('abort')
+}
 </script>
 
 <template>
@@ -42,7 +48,8 @@ function onCtrlEnter(evt: KeyboardEvent) {
     <Textarea v-model="text" @keydown="onCtrlEnter" placeholder="Type a message" />
 
     <div class="actions">
-      <Button @click="send" :disabled="!canSend">Send</Button>
+      <Button v-if="isProcessing" @click="handleAbort">Abort</Button>
+      <Button v-else @click="send" :disabled="!canSend">Send</Button>
     </div>
   </div>
 </template>
diff --git a/src/components/Chat/ChatRoom.vue b/src/components/Chat/ChatRoom.vue
index 3ccdcf4..d57b97a 100644
--- a/src/components/Chat/ChatRoom.vue
+++ b/src/components/Chat/ChatRoom.vue
@@ -14,15 +14,15 @@ export type ChatRoomEmits = ChatInputEmits & {
   renameTitle: [newTitle: string]
 }
 
-const props = defineProps<ChatRoomProps>()
+defineProps<ChatRoomProps>()
 
 const emit = defineEmits<ChatRoomEmits>()
 
-async function handleSend(content: string) {
+function handleSend(content: string) {
   emit('send', content)
 }
 
-async function renameChat(newName: string) {
+function renameChat(newName: string) {
   emit('renameTitle', newName)
 }
 </script>
@@ -33,7 +33,7 @@ async function renameChat(newName: string) {
 
     <ChatMessageList :messages="messages" />
 
-    <ChatInput :is-processing="isProcessing" @send="handleSend" />
+    <ChatInput :is-processing="isProcessing" @send="handleSend" @abort="emit('abort')" />
   </div>
 </template>
 
diff --git a/src/pages/chat.vue b/src/pages/chat.vue
index 3702282..4078ff5 100644
--- a/src/pages/chat.vue
+++ b/src/pages/chat.vue
@@ -4,6 +4,7 @@ import { useAsyncData, useLoading } from '@0x-jerry/vue-kit'
 import { getCurrentWindow } from '@tauri-apps/api/window'
 import { debounce } from 'lodash-es'
 import OpenAI from 'openai'
+import { ChatCompletionStream } from 'openai/lib/ChatCompletionStream.mjs'
 import { ChatCompletionMessageParam } from 'openai/resources/index.mjs'
 import { onMounted, reactive, ref } from 'vue'
 import AutoResizeContainer from '../components/AutoResizeContainer.vue'
@@ -31,6 +32,8 @@ const state = reactive({
   responseMsg: null as Optional<IChatHistoryMsgModel>,
 })
 
+let streamRef: Optional<ChatCompletionStream>
+
 const chatHistory = ref<IChatHistoryModel>()
 
 const promptConfigApi = useAsyncData(promptConfigTable.getById)
@@ -174,6 +177,8 @@ async function startChatStream(msgs: IChatHistoryMsgItem[]) {
     }),
   })
 
+  streamRef = stream
+
   stream.on('message', (msg) => {
     let content = ''
 
@@ -210,6 +215,7 @@ async function startChatStream(msgs: IChatHistoryMsgItem[]) {
 
   stream.on('end', () => {
     state.responseMsg = null
+    streamRef = null
 
     resultPromise.resolve()
   })
@@ -220,15 +226,20 @@ async function startChatStream(msgs: IChatHistoryMsgItem[]) {
 
   return resultPromise.promise
 }
+
+function handleAbort() {
+  streamRef?.abort()
+}
 </script>
 
 <template>
   <AutoResizeContainer :width="400">
     <div class="page bg-white">
-      <ChatPageHead :icon="promptConfigApi.data.value?.icon"  :title="promptConfigApi.data.value?.name" />
-      
+      <ChatPageHead :icon="promptConfigApi.data.value?.icon" :title="promptConfigApi.data.value?.name" />
+
       <template v-if="chatHistory">
-        <ChatRoom :title="chatHistory.name" :messages="state.messages" :is-processing="handleSendMsg.isLoading" @rename-title="updateChatTitle" @send="handleSendMsg" />
+        <ChatRoom :title="chatHistory.name" :messages="state.messages" :is-processing="handleSendMsg.isLoading"
+          @rename-title="updateChatTitle" @send="handleSendMsg" @abort="handleAbort" />
       </template>
     </div>
   </AutoResizeContainer>
